### DISCUSSION TITLE
How can <link rel="stylesheet" href="styles.css"> not work with live-server?

### DISCUSSION THREAD

**Discussion by JNewstedt** [tailwindlabs/tailwindcss#5660] [Help]
I'm going through a tailwind tutorial by The Net Ninja (link: https://youtu.be/3ZMUgga6SsY) and I'm up to where I add this just above the closing `</head>` tag:

`<link rel="stylesheet" href="styles.css">`

I'm using VS Code and running a live-server through the terminal. Every file is in the right place and changes are updated as I make them, so I know there's a connection there. However, the external css link is just not being recognized. None of my css classes are working. I'm obviously missing something, but I cannot figure out what that is.

Here is my setup and index file:

![tailwindCSS](https://user-images.githubusercontent.com/91703537/135539542-2cc021f1-ed9c-463a-a400-aecf3843db65.jpg)

**Comment by simonswiss** [comment]
Hey!

Are you able to share the code you've got in a GitHub public repo?

What's in your `public/styles.css` file, can you see the Tailwind CSS classes in there? That file should be the _generated_ CSS, not the `@tailwind base/components/utilities` directives (which go in `src/styles.css`)

I assume you're running `live-server` from the `public` directory, with `live-server public`? If you need more help, please share some code 👍

**Comment by JNewstedt** [comment]
Sorry, I'm still having this problem. 

Here's the layout of my site as seen in VS Code:
![styles](https://user-images.githubusercontent.com/91703537/139948336-bf1b27a4-0927-4b55-a0ce-154da5a95c64.jpg)


My `public/styles.css` file was generated after installing Tailwind. The steps I took to do that are as follows:
**1.**	I opened terminal in VS Code (using powershell) 
**2.**	Installed _package.json_ by typing _npm init -y_
**3.**	Installed tailwind by typing: _npm install tailwindcss_
(This created “package-lock.json” and the folder that houses the tailwind package “node_modules”)
**4.**	Then I created a _src_ folder and a _public_ folder for my site.
**5.**	Inside the _src_ folder I created a file and named it _styles.css_ and added this to call up Tailwind:
```
@import "tailwindcss/base";
@import "tailwindcss/components";
@import "tailwindcss/utilities"; 
```
**6.**	In the “package.json” file, I deleted the following line under "scripts:"
`"test": "echo \"Error: no test specified\" && exit 1"`
and replaced it with 
`"build-css": "tailwindcss build -i src/styles.css -o public/styles.css" `
so that anything I made would be built in the site folder.

I'd love to know why the index.htm file is just not recognizing the styles.css file. I can't see how it isn't. 

Additionally, if anyone knows of an easier way to generate Tailwind for a site, I'm open to any suggestions! I'm only learning this recently.

**Comment by NovusOpus** [comment]
You can check this [video](https://www.youtube.com/watch?v=DxcJbrs6rKk), the section you want to check starts at [00:14](https://www.youtube.com/watch?v=DxcJbrs6rKk&t=14s), here @simonswiss explains and show you how to setup your project and build your CSS file with Tailwind CLI.

---

**Alternatively If you want to make it work like in your config:**

Edit your `package.json` file like this:

```JSON
{
  "dependencies": {
    "tailwindcss": "^2.2.19"
  },
  "scripts": {
    "build": "tailwindcss -o ./public/styles.css --jit --purge './**/*.html'"
  }

}
```
Or if you want it to watch and update your build CSS file at every time you save your HTML file after adding more classes:
```JSON
{
  "dependencies": {
    "tailwindcss": "^2.2.19"
  },
  "scripts": {
    "build": "tailwindcss -o ./public/styles.css --jit --purge './**/*.html' -w "
  }

}
```
Now in those examples I'm following your files as reference, but note that you don't have to generate 2 different CSS files following this approach since Tailwind CLI is already generating the final file for you so you don't have to "Call Tailwind" using:

```CSS
@import "tailwindcss/base";
@import "tailwindcss/components";
@import "tailwindcss/utilities";
```
---
**And if you want to make use of the **Tailwind Configuration**, Like for example extending classes or making changes to Tailwind Utilities:**

Create a new `tailwind.config.js` file writing: `npx tailwindcss init` in your console, This will generate this file:
```JS
// tailwind.config.js
module.exports = {
  purge: [],
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {},
  },
  variants: {},
  plugins: [],
}
```
Now edit that file to look like this: 
```JS
// tailwind.config.js
module.exports = {
  mode: "jit",
  purge: ["./**/*.html"],
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {
      colors: {
        sky: colors.sky,
        cyan: colors.cyan,
      },
    },
  },
  variants: {},
  plugins: [],
}
```
and update your package.json:
```JSON
{
  "dependencies": {
    "tailwindcss": "^2.2.19"
  },
  "scripts": {
    "build": "tailwindcss -o ./public/styles.css  -w "
  }

}
```
Notice how we removed `--jit --purge './**/*.html'` since we added those flags into the `tailwind.config.js`, so when running: `npm build` , or `yarn build`  Tailwind CLI will look at your `tailwind.config.js` file and apply your settings or changes to your build CSS file.

---
Also Now there's even a faster and easy way: [Tailwind Just-in-Time CDN](https://www.youtube.com/watch?v=QAIdYGslQGo). Currently in Alpha so please note:

> The Just-in-Time CDN is intended for development purposes only, and should not be used in production.

**Reply by JNewstedt** [reply]
@novuspace, thanks for the info! The first @simonswiss video has a clean integration I knew nothing about before now. I'm looking forward to the stable version of 3.0 and its JIT CDN! 

Thanks again.
